### PR TITLE
(embassy-stm32): uart flowcontrol

### PR DESCRIPTION
**Pull request by MathiasKoch**
Add RTS & CTS flow control to stm32 UARTs

### CODE REVIEW

File: embassy-stm32/src/usart/mod.rs [hunk 1]
```diff
@@ -241,6 +240,29 @@ impl<'d, T: BasicInstance, TxDma, RxDma> Uart<'d, T, TxDma, RxDma> {
...
+        tx_dma: impl Peripheral<P = TxDma> + 'd,
+        rx_dma: impl Peripheral<P = RxDma> + 'd,
+        config: Config,
+    ) -> Self {
+        into_ref!(cts, rts);
+
+        unsafe {
+            rts.set_as_af(rts.af_num(), AFType::OutputPushPull);
+            cts.set_as_af(cts.af_num(), AFType::Input);
+            T::regs().cr3().write(|w| {
```
Comment (Dirbaio) [inline embassy-stm32/src/usart/mod.rs:258]: Are you sure this works? `new()` is doing RCC reset+enable *after* this write, so it looks a bit suspicious.

Comment (MathiasKoch) [reply embassy-stm32/src/usart/mod.rs:258]: Hmm, I am not sure that I see how RCC reset and enable alters the UART control regs?

Comment (MathiasKoch) [reply embassy-stm32/src/usart/mod.rs:258]: I will be able to verify this on monday though.

Comment (Dirbaio) [reply embassy-stm32/src/usart/mod.rs:258]: Writes to peripheral regs simply don't work if the peripehral is not rcc-enabled, and the rcc-reset clears all the registers.

**Comment by MathiasKoch** [comment]
@Dirbaio Took a long time for me to get around to resolving this, but it should be good to go now.

**Review by Dirbaio — APPROVED** [review]
Cool, thank you! 

bors r+

**Pull request merged**